api/images.go

```diff
@@ -0,0 +1,163 @@
+		log.Errorf("Error getting K8s config: %s", err.Error())
+		c.JSON(http.StatusBadRequest, pkgCommmon.ErrorResponse{
+			Code:    http.StatusBadRequest,
+			Message: "Error getting K8s config",
+			Error:   err.Error(),
+		})
+		return
+	}
+
+	imageList, err := listAllImages(client, "")
+	if err != nil {
+		err := errors.Wrap(err, "Error during request processing")
+		log.Error(err.Error())
+		httpStatusCode := http.StatusInternalServerError
+		c.JSON(httpStatusCode, pkgCommmon.ErrorResponse{
+			Code:    httpStatusCode,
+			Message: "Error getting Pods",
+			Error:   err.Error(),
+		})
+		return
+	}
+	c.JSON(http.StatusOK, imageList)
+}
+
+// GetDeploymentImages list all used images in deployment
+func GetDeploymentImages(c *gin.Context) {
+	release := c.Param("name")
+	log.Infof("getting images for deployment: [%s]", release)
+
+	kubeConfig, ok := GetK8sConfig(c)
+	if !ok {
+		return
+	}
+
+	client, err := helm.GetK8sConnection(kubeConfig)
+	if err != nil {
+		log.Errorf("Error getting K8s config: %s", err.Error())
+		c.JSON(http.StatusBadRequest, pkgCommmon.ErrorResponse{
+			Code:    http.StatusBadRequest,
+			Message: "Error getting K8s config",
+			Error:   err.Error(),
+		})
+		return
+	}
+
+	selector := fmt.Sprintf("release=%s", release)
+	log.Infof("Label selector: %s", selector)
+
+	imageList, err := listAllImages(client, selector)
+	if err != nil {
+		err := errors.Wrap(err, "Error during request processing")
+		log.Error(err.Error())
+		httpStatusCode := http.StatusInternalServerError
+		c.JSON(httpStatusCode, pkgCommmon.ErrorResponse{
+			Code:    httpStatusCode,
+			Message: "Error getting Pods",
+			Error:   err.Error(),
+		})
+		return
+	}
+
+	c.JSON(http.StatusOK, imageList)
+}
+
+func listAllImages(client *kubernetes.Clientset, labelSelector string) ([]apiclient.ClusterImage, error) {
+	var err error
+	var podList []v1.Pod
+	podList, err = listPods(client, "", labelSelector)
+	if err != nil {
+		return nil, err
+	}
+
+	imageList := make([]apiclient.ClusterImage, 0)
+	for _, pod := range podList {
+		images := getPodImages(pod)
+		imageList = append(imageList, images...)
+	}
+	removeDuplicatedImages(&imageList)
+	return imageList, nil
+}
+
+func getPodImages(pod v1.Pod) []apiclient.ClusterImage {
+
+	images := make([]apiclient.ClusterImage, 0)
+	for _, container := range pod.Status.ContainerStatuses {
+		fullName := strings.Split(container.Image, ":")
+		var name string
+		tag := "latest"
+		if len(fullName) > 1 {
+			name = fullName[0]
+			tag = fullName[1]
+		}
+		fullDigest := strings.Split(container.ImageID, "@")
+		var digest string
+		if len(fullDigest) > 1 {
+			digest = fullDigest[1]
+		} else {
+			continue
+		}
+
+		image := apiclient.ClusterImage{
+			ImageName:   name,
+			ImageTag:    tag,
+			ImageDigest: digest,
+		}
+		images = append(images, image)
+	}
+	return images
+}
+
+func removeDuplicatedImages(images *[]apiclient.ClusterImage) {
+	found := make(map[string]bool)
+	j := 0
+	for i, image := range *images {
```

Comment:
The loop will make a copy of each item when iterating through 'images'. Use '[]*apiclient.ClusterImage ' to avoid that.